examples/src/utils.js

```diff
@@ -0,0 +1,322 @@
+  if (!companionVastTracker) {
+    return;
+  }
+  companionVastTracker.trackImpression();
+};
+
+/**
+ * Send complete tracker using the vastTracker
+ */
+const handleCompleteTrackers = () => {
+  if (!vastTracker) {
+    return;
+  }
+  vastTracker.complete();
+};
+
+/**
+ * Send pause tracker using the vastTracker
+ */
+
+const handlePauseTrackers = () => {
+  if (!vastTracker) {
+    return;
+  }
+  vastTracker.setPaused(true);
+};
+
+/**
+ * Send resume tracker using the vastTracker
+ */
+const handleResumeTrackers = () => {
+  if (!vastTracker) {
+    return;
+  }
+  vastTracker.setPaused(false);
+};
+
+/**
+ * Send progress tracker(firstQuartile, midpoint...) using the vastTracker
+ */
+const handleTimeUpdateTrackers = () => {
+  if (!vastTracker) {
+    return;
+  }
+  vastTracker.setProgress(videoPlayer.currentTime);
+};
+
+/**
+ * Send error trackers using the vastTracker
+ */
+
+const handleErrorTrackers = () => {
+  if (!vastTracker) {
+    return;
+  }
+  vastTracker.error();
+};
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////COMPANION HANDLING////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * @returns the first valid companion creative
+ */
+const getCreativeCompanion = (ad) => {
+  if (!ad.creatives) {
+    return null;
+  }
+
+  const validCreative = ad.creatives.find(
+    (creative) => creative.type === 'companion'
+  );
+
+  return validCreative || null;
+};
+
+/**
+ * @returns an Object containing the image url and the variation
+ */
+
+const getVariationData = (ad) => {
+  if (!ad.creatives) {
+    return null;
+  }
+  const validCreative = getCreativeCompanion(ad);
+
+  if (!validCreative.variations.length) {
+    return null;
+  }
+
+  return {
+    imgUrl: getVariationUrl(validCreative.variations[0]),
+    variation: validCreative.variations[0],
+  };
+};
+
+/**
+ * @returns the image url of the variation to display
+ */
+
+const getVariationUrl = (variation) => {
+  if (!variation.staticResources.length) {
+    return null;
+  }
+
+  return variation.staticResources[0].url;
+};
+
+const displayVariation = (variationData) => {
+  const { imgUrl, variation } = variationData;
+  if (!variation) {
+    return null;
+  }
+  variationImg.src = imgUrl;
+  variationImg.style.height = variation.assetHeight + 'px';
+  variationImg.style.assetWIdth = variation.assetWidth + 'px';
+  variationContainer.style.width = variation.width + 'px';
+  variationContainer.style.height = variation.height + 'px';
+  variationImg.style.maxWidth = variation.expandedWidth + 'px';
+  variationImg.style.maxHeight = variation.expandedHeight + 'px';
+};
+
+let prerollPlayed = false;
+
+videoPlayer.addEventListener('play', () => {
+  if (!prerollPlayed) {
+    playPreroll();
+    prerollPlayed = true;
+  }
+});
```

Comment:
Do we need this listener if it's already set in examples/src/index.js ?

Reply:
No, definitly something i forgot to remove :)